test/conftest.py

```diff
@@ -0,0 +1,18 @@
+import os
+import tempfile
+
+import pytest
+
+# whether to remove storage files after a test has finished
+REMOVE_STORE = True
+
+
+@pytest.fixture
+def hdf5_file():
+    """Generate a temporary hdf5 file."""
+    store_file = tempfile.mkstemp(suffix=".hdf5")[1]
+    try:
+        yield store_file
+    finally:
+        if REMOVE_STORE and os.path.exists(store_file):
+            os.remove(store_file)
```

Comment:
I guess this is to avoid extra indentation by a tempfile context?

Could generalize that and pass the suffix as argument. Can also be used in other places then.